src/libraries/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs

```diff
@@ -877,6 +882,59 @@ public void SendAsync_SuppressedGlobalStaticPropagationEnvVar(string envVarValue
             }, UseVersion.ToString(), TestAsync.ToString(), envVarValue).Dispose();
         }
 
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsNotBrowser))]
+        [MemberData(nameof(UseSocketsHttpHandler_WithIdFormat_MemberData))]
+        public async Task SendAsync_HeadersAreInjectedOnRedirects(bool useSocketsHttpHandler, ActivityIdFormat idFormat)
+        {
+            Activity parent = new Activity("parent");
+            parent.SetIdFormat(idFormat);
+            parent.TraceStateString = "Foo";
+            parent.Start();
+
+            await GetFactoryForVersion(UseVersion).CreateServerAsync(async (originalServer, originalUri) =>
+            {
+                await GetFactoryForVersion(UseVersion).CreateServerAsync(async (redirectServer, redirectUri) =>
+                {
+                    Task clientTask = GetAsync(UseVersion.ToString(), TestAsync.ToString(), originalUri, useSocketsHttpHandler: useSocketsHttpHandler);
+
+                    Task<HttpRequestData> serverTask = originalServer.HandleRequestAsync(HttpStatusCode.Redirect, new[] { new HttpHeaderData("Location", redirectUri.AbsoluteUri) });
+
+                    await Task.WhenAny(clientTask, serverTask);
+                    Assert.False(clientTask.IsCompleted, $"{clientTask.Status}: {clientTask.Exception}");
+                    HttpRequestData firstRequestData = await serverTask;
+                    AssertHeadersAreInjected(firstRequestData, parent);
+
+                    serverTask = redirectServer.HandleRequestAsync();
+                    await TestHelper.WhenAllCompletedOrAnyFailed(clientTask, serverTask);
+                    HttpRequestData secondRequestData = await serverTask;
+                    AssertHeadersAreInjected(secondRequestData, parent);
+
+                    // Tests that change the default should use RemoteExecutor
+                    Assert.Same(TextMapPropagator.CreateDefaultPropagator(), TextMapPropagator.Current);
```

Comment:
I wouldnt' make this to assert. instead convert it to if statement and exit the test if it is not teh default.

Reply:
Why? It should never be the case that they are not the same unless a rogue test changes it outside of RemoteExecutor.

If current would ever not be `CreateDefaultPropagator` by default, this test should be changed to use `RemoteExecutor` instead of silently exiting.

Reply:
I would say it is not right to assume the global state value. Imagin this fail, it would be very difficult to diagnose the failure. the other idea could be run your test with remote execution and force the default propagator in the beginning.

Reply:
It's part of the propagators API contract that the value of `TextMapPropagator.Current` is `CreateDefaultPropagator` at the start of the process. If a regular test changed its value, it would impact a number of other tests, besides just this one, that may implicitly rely on the default behavior.
The assert here catches that and will only trip if a test in the Http functional tests assembly does that. It should be easy to spot any recent additions of `TextMapPropagator.Current = xyz`.

If a test changes global state, we always run those in RemoteExecutor instead of isolating the majority of other tests.